src/BloomExe/web/controllers/BookSettingsApi.cs

```diff
@@ -54,12 +60,36 @@ private  void HandleBookSettings(ApiRequest request)
 
 		private bool GetIsBookATemplate()
 		{
-			return false;
+			if (_bookSelection == null || _bookSelection.CurrentSelection == null || _bookSelection.CurrentSelection.BookInfo == null)
+				return false;
+			var info = _bookSelection.CurrentSelection.BookInfo;
+			SaveOriginalTypeIfNecessary(info);
+			return info.Type == Book.Book.BookType.Template;
 		}
 
 		private void UpdateBookTemplateMode(bool isTemplateBook)
 		{
-			
+			if (_bookSelection == null || _bookSelection.CurrentSelection == null || _bookSelection.CurrentSelection.BookInfo == null)
+				return;     // sheer paranoia
+			var info = _bookSelection.CurrentSelection.BookInfo;
+			SaveOriginalTypeIfNecessary(info);
+			if (isTemplateBook)
+			{
+				info.Type = Book.Book.BookType.Template;
+			}
+			else
+			{
+				Book.Book.BookType type = _bookOriginalType[info];
+				if (type == Book.Book.BookType.Template)
+					type = Book.Book.BookType.Publication;
```

Comment:
I'm having trouble reasoning about this... I think if you have a booktype.shell, make a template, re-run bloom, and turn off the template switch, you now go to "publication". I need to look more into what those terms even mean.

I'm also wondering if this "prior state" belongs in the book rather than here. For one thing it could be persisted. But i'm not sure and its Friday evening...